File: android/app/src/main/java/com/screenmeet/sdkdemo/ConfidentialityActivity.java
package com.screenmeet.sdkdemo;

import android.app.Activity;
import android.os.Bundle;
import android.util.DisplayMetrics;
import android.view.View;
import android.view.ViewGroup;
import android.widget.HorizontalScrollView;
import android.widget.LinearLayout;
import android.widget.Space;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.screenmeet.sdk.ScreenMeet;

import java.util.ArrayList;
import java.util.Random;

public class ConfidentialityActivity extends AppCompatActivity {

    private final ArrayList<View> viewToObfuscate = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_ui_confidentiality);

        findViewById(R.id.obfuscateNew).setOnClickListener(v -> {
            View obfuscatedView = constructConfidentialView(ConfidentialityActivity.this, viewToObfuscate.size());

            ((ViewGroup)findViewById(R.id.obfuscateContainer)).addView(obfuscatedView,
                    new ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT));
            viewToObfuscate.add(obfuscatedView);
            ScreenMeet.setConfidential(obfuscatedView.findViewWithTag("tv"));
        });

        findViewById(R.id.deobfuscateNew).setOnClickListener(v -> {
            if(!viewToObfuscate.isEmpty()){
                View view = viewToObfuscate.get(viewToObfuscate.size() - 1);
                ScreenMeet.unsetConfidential(view.findViewWithTag("tv"));
                viewToObfuscate.remove(view);
                ((ViewGroup)findViewById(R.id.obfuscateContainer)).removeView(view);
            }
        });
    }

    public static View constructConfidentialView(Activity context, int id){
        DisplayMetrics displayMetrics = new DisplayMetrics();
        context.getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
        int width = displayMetrics.widthPixels;

        TextView view = new TextView(context);
        view.setTag("tv");
        String text = "Secret text " + id;
        view.setText(text);
        int textSize = new Random().nextInt(25) + 11;
        view.setTextSize(textSize);
        int measuredSize = textSize * (view.getText().length() + 5);

        Space spaceLeft = new Space(context);
        Space spaceRight = new Space(context);

        LinearLayout container = new LinearLayout(context);
        container.addView(spaceLeft, new ViewGroup.LayoutParams(width - measuredSize, ViewGroup.LayoutParams.MATCH_PARENT));
        container.addView(view, new ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT));
        container.addView(spaceRight, new ViewGroup.LayoutParams(width - measuredSize, ViewGroup.LayoutParams.MATCH_PARENT));

        HorizontalScrollView scrollView = new HorizontalScrollView(context);
        scrollView.setHorizontalScrollBarEnabled(false);
        scrollView.addView(container, new ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT));

        return scrollView;
    }
}

File: android/app-kotlin/src/main/java/com/screenmeet/sdkkotlindemo/ConfidentialityActivity.kt
package com.screenmeet.sdkkotlindemo

import android.app.Activity
import android.os.Bundle
import android.util.DisplayMetrics
import android.view.View
import android.view.ViewGroup
import android.widget.HorizontalScrollView
import android.widget.LinearLayout
import android.widget.Space
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.screenmeet.sdk.ScreenMeet
import java.util.*

class ConfidentialityActivity : AppCompatActivity() {
    private val viewToObfuscate = ArrayList<View>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_ui_confidentiality)

        findViewById<View>(R.id.obfuscateNew).setOnClickListener {
            val obfuscatedView = constructConfidentialView(this@ConfidentialityActivity, viewToObfuscate.size)
            (findViewById<View>(R.id.obfuscateContainer) as ViewGroup).addView(obfuscatedView,
                    ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT))
            viewToObfuscate.add(obfuscatedView)
            ScreenMeet.appStreamVideoSource().setConfidential(obfuscatedView.findViewWithTag("tv"))
        }

        findViewById<View>(R.id.deobfuscateNew).setOnClickListener {
            if (viewToObfuscate.isNotEmpty()) {
                val view = viewToObfuscate[viewToObfuscate.size - 1]
                ScreenMeet.appStreamVideoSource().unsetConfidential(view.findViewWithTag("tv"))
                viewToObfuscate.remove(view)
                (findViewById<View>(R.id.obfuscateContainer) as ViewGroup).removeView(view)
            }
        }
    }

    companion object {
        fun constructConfidentialView(context: Activity, id: Int): View {
            val displayMetrics = DisplayMetrics()
            context.windowManager.defaultDisplay.getMetrics(displayMetrics)
            val width = displayMetrics.widthPixels
            val view = TextView(context)

            view.tag = "tv"
            val text = "Secret text $id"
            view.text = text
            val textSize = Random().nextInt(25) + 11
            view.textSize = textSize.toFloat()
            val measuredSize = textSize * (view.text.length + 5)
            val spaceLeft = Space(context)
            val spaceRight = Space(context)

            val container = LinearLayout(context)
            container.addView(spaceLeft, ViewGroup.LayoutParams(width - measuredSize, ViewGroup.LayoutParams.MATCH_PARENT))
            container.addView(view, ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT))
            container.addView(spaceRight, ViewGroup.LayoutParams(width - measuredSize, ViewGroup.LayoutParams.MATCH_PARENT))

            val scrollView = HorizontalScrollView(context)
            scrollView.isHorizontalScrollBarEnabled = false
            scrollView.addView(container, ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT))
            return scrollView
        }
    }
}